ui/apps/platform/src/Containers/NetworkGraph/utils/modelUtils.tsx

```diff
@@ -16,19 +27,56 @@ function getNameByEntity(entity: NetworkEntityInfo): string {
     }
 }
 
+function getDataByEntityType(entity: NetworkEntityInfo, policyIds: string[]): CustomNodeData {
+    if (entity.type === 'DEPLOYMENT') {
+        const data: DeploymentData = {
+            ...entity,
+            policyIds,
+        };
+        return data;
+    }
+    if (entity.type === 'INTERNET') {
+        const data: ExternalEntitiesData = {
+            ...entity,
+            type: 'EXTERNAL_ENTITIES',
+        };
+        return data;
+    }
+    const data: CIDRBlockData = {
+        ...entity,
+        type: 'CIDR_BLOCK',
+    };
+    return data;
```

Comment:
nit: Would it be more clear to make `DEPLOYMENT` the default return type ?

Reply:
Hmm, yea we can do that. I wanted to actually make the default be either throwing an error or returning another value, but Typescript was acting up because of that

Reply:
Speaking of `type.utils.ts`, you could use the `ensureExhaustive` function with a switch/case here to make it so that it is impossible for the app to compile if it would hit the default case:
```typescript
function getDataByEntityType(entity: NetworkEntityInfo, policyIds: string[]): CustomNodeData {
    switch (entity.type) {
        case 'DEPLOYMENT':
            return { ...entity, policyIds };
        case 'INTERNET':
            return { ...entity, type: 'EXTERNAL_ENTITIES' };
        case 'EXTERNAL_SOURCE':
            return { ...entity, type: 'CIDR_BLOCK' };
        default:
            return ensureExhaustive(entity);
    }
}
```

If you miss a case, or if the `NetworkEntityInfo["type"]` field gets an additional value down the line, the default case here will result in a type error.

Reply:
Oh that's really useful! Let me try that out as well